plugins/extractors/date/date.go

```diff
@@ -0,0 +1,81 @@
+package date
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	"github.com/odpf/meteor/core/extractor"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/proto/odpf/meta/facets"
+	"time"
+)
+
+const (
+	MaxDates = 3
+)
+
+type Extractor struct {
+	logger plugins.Logger
+}
+
+type Payload struct{
+	Value string
+}
+
+type Payloads []Payload
+
+func (p *Payload) ToJSON() ([]byte, error) {
+	return json.Marshal(p)
+}
+
+func (p *Payload) ToProto() ([]byte, error) {
+	protoC := &facets.Custom{CustomProperties: map[string]string{
+		"Value": p.Value,
+	}}
+	return proto.Marshal(protoC)
+}
+
+func (d *Extractor) Extract(ctx context.Context, config map[string]interface{}, out chan<- interface{}) (err error) {
```

Comment:
Using go channels approach will increase complexity for end-users to write extractors. Ideally, we want the extractor to be very dumb and easy to implement.

Reply:
IMHO we could probably abstract it but even then not sure how much simpler it would be. 🤔

Reply:
@ravisuhag  btw this would become what we are avoiding in the first place right, allowing them to output dynamic data.

Reply:
Each sink will do a hard check of the type if needed. So we can keep it this way.